docs/advanced/tls_porting.md

```diff
@@ -14,61 +14,77 @@ Creating session keys is only one use for random values; they have far more comp
 
-- ``size_t len``: The length of the output buffer. The function shouldn't write more data than this to the output buffer under any circumstances.
+    **Warning:** Although it is possible to fill this buffer without a strong hardware entropy source, we strongly advise against it, because it will nullify any security provided by mbed TLS.
 
-- ``size_t *olen``: The length of the data written into the output buffer. It tells the caller how much entropy has been collected and how many bytes of the output buffer it can use. It should always reflect the exact amount of entropy collected; setting it higher than the actual number of bytes collected is a serious security risk.
+- ``size_t length``: The length of the output buffer. The function shouldn't write more data than this to the output buffer under any circumstances.
 
-### Setting the macros
+- ``size_t *output_length``: The length of the data written into the output buffer. It tells the caller how much entropy has been collected and how many bytes of the output buffer it can use. It should always reflect the exact amount of entropy collected; setting it higher than the actual number of bytes collected is a serious security risk.
 
-To register your `mbedtls_hardware_poll` function with the mbed TLS entropy framework, you have to add `MBEDTLS_ENTROPY_HARDWARE_ALT` to your macros in `targets.json`:
 
-```
-"macros": ["MBEDTLS_ENTROPY_HARDWARE_ALT", etc.]
-```
+### Indicating the presence of a TRNG
 
-You can read more about how to add a macro for your target [here](../mbed_OS/Targets.md).
+To indicate that the target has an entropy source, you have to add `TRNG` to the capabilities of the target in `targets.json`:
 
-<span class="notes">**Note:** You should define the `MBEDTLS_NO_PLATFORM_ENTROPY` macro on any platform that does not support standards like /dev/urandom or Windows CryptoAPI.</span>
+```
+"device_has": ["TRNG", etc.]
+```
 
 ## 4. How to implement the Non-Volatile seed entropy source
 
-If a hardware platform does not have a hardware entropy source to leverage into the entropy pool, alternatives have to be considered. As said before, a strong entropy source is crucial for security of cryptographic and TLS operations. For platforms that support non-volatile memory, an option is to use the NV seed entropy source that is provided with mbed TLS.
+If a hardware platform does not have a hardware entropy source to leverage into the entropy pool, alternatives have to be considered. As stated above, a strong entropy source is crucial for security of cryptographic and TLS operations. For platforms that support non-volatile memory, an option is to use the NV seed entropy source that is provided with mbed TLS.
+
+This makes mbed TLS use a fixed amount of entropy as a seed and update this seed each time it runs.
```

Comment:
I do not think `entropy as a seed and update this seed each time it runs` is clear enough. Does this mean whenever the device is reset, the entropy source is polled, etc.